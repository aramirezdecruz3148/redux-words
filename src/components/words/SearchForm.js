import React from 'react';
import PropTypes from 'prop-types';

function SearchForm({ handleOnSubmit, searchTerm, handleOnChange }) {
  return (
    <form onSubmit={handleOnSubmit}>
      <input 
        type='text' 
        value={searchTerm} 
        name='searchTerm' 
        onChange={handleOnChange} 
      />\
      <button>Search üîç</button>
    </form>
  );
}

SearchForm.propTypes = {
  handleOnSubmit: PropTypes.func.isRequired,
  searchTerm: PropTypes.string.isRequired,
  handleOnChange: PropTypes.func.isRequired
};

export default SearchForm;
