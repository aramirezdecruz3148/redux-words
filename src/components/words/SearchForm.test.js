import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm component', () => {
  it('renders SearchForm', () => {
    const wrapper = shallow(<SearchForm searchTerm='hi' handleOnChange={() => { }}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
