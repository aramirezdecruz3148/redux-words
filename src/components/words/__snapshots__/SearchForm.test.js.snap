// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component renders SearchForm 1`] = `
<input
  name="searchTerm"
  onChange={[Function]}
  placeholder="🔍"
  type="text"
  value="hi"
/>
`;
